pkg/operation/botanist/controlplane.go

```diff
@@ -1332,20 +1332,19 @@ func (b *Botanist) DeployKubeScheduler(ctx context.Context) error {
 }
 
 // DefaultKubeAPIServerService returns a deployer for kube-apiserver service.
-func (b *Botanist) DefaultKubeAPIServerService() component.DeployWaiter {
-	var sniService *client.ObjectKey
-
-	if b.APIServerSNIEnabled() {
-		sniService = &client.ObjectKey{Name: common.IstioIngressGatewayServiceName, Namespace: common.IstioIngressGatewayNamespace}
-	}
+func (b *Botanist) DefaultKubeAPIServerService(sniPhase component.Phase) component.DeployWaiter {
+	return b.kubeAPIServiceService(sniPhase)
+}
 
+func (b *Botanist) kubeAPIServiceService(sniPhase component.Phase) component.DeployWaiter {
```

Comment:
What's the purpose of this function? The only thing `DefaultKubeAPIServerService` is calling is this one, so it can probably be moved into `DefaultKubeAPIServerService` again?

Reply:
It's called by `DeployKubeAPIService` as well